pkg/interface/src/components/Omnibox.js

```diff
@@ -0,0 +1,251 @@
+            result.link.toLowerCase().includes(query) ||
+            result.app.toLowerCase().includes(query) ||
+            (result.host !== null ? result.host.includes(query) : false)
+          );
+        })
+      );
+    });
+
+    this.setState({ results: results });
+  }
+
+    setPreviousSelected() {
+      const current = this.state.selected;
+      const flattenedResults = Array.from(this.state.results.values()).flat();
+      const totalLength = flattenedResults.length;
+      if (current !== '') {
+        const currentIndex = flattenedResults.indexOf(
+          ...flattenedResults.filter((e) => {
+            return e.link === current;
+          })
+        );
+        if (currentIndex > 0) {
+        const nextLink = flattenedResults[currentIndex - 1].link;
+        this.setState({ selected: nextLink });
+        } else {
+        const nextLink = flattenedResults[totalLength - 1].link;
+        this.setState({ selected: nextLink });
+        }
+      } else {
+        const nextLink = flattenedResults[totalLength - 1].link;
+        this.setState({ selected: nextLink });
+      }
+  }
+
+  setNextSelected() {
+    const current = this.state.selected;
+    const flattenedResults = Array.from(this.state.results.values()).flat();
+    if (current !== '') {
+      const currentIndex = flattenedResults.indexOf(
+        ...flattenedResults.filter((e) => {
+          return e.link === current;
+        })
+      );
+      if (currentIndex < flattenedResults.length - 1) {
+      const nextLink = flattenedResults[currentIndex + 1].link;
+      this.setState({ selected: nextLink });
+      } else {
+      const nextLink = flattenedResults[0].link;
+      this.setState({ selected: nextLink });
+      }
+    } else {
+      const nextLink = flattenedResults[0].link;
+      this.setState({ selected: nextLink });
+    }
+  }
+
+  render() {
+    const { props, state } = this;
+    const categoryResult = [];
+
+    const renderResults = <Box maxHeight="400px" overflowY="scroll" overflowX="hidden">
+      {categoryResult}
+    </Box>;
+
+      ['commands', 'subscriptions', 'groups', 'apps'].map((category, i) => {
+      const categoryResults = state.results.get(category);
+      if (categoryResults.length > 0) {
+        const each = categoryResults.map((result, i) => {
+          return <OmniboxResult
+            key={i}
+            icon={result.app}
+            text={result.title}
+            subtext={cite(result.host)}
+            link={result.link}
+            navigate={() => this.navigate(result.link)}
+            selected={this.state.selected}
+            dark={props.dark}
+                 />;
+        });
+         categoryResult.push(<Box key={i} width='max(50vw, 300px)' maxWidth='600px'>
+        <Rule borderTopWidth="0.5px" color="washedGray" />
+        <Text gray ml={2}>{category.charAt(0).toUpperCase() + category.slice(1)}</Text>
+          {each}
+        </Box>);
+      }
+    });
+
+    return (
+      <>
```

Comment:
What is the advantage of wrapping the Box in this?

Reply:
I think that's from when I was working with Gavin; when you do that it's short hand for `<Fragment>` so it doesn't wrap it in a div or the like. But because we have the Box anyway, it's moot, let me double check and push